src/gluonts/shell/train.py

```diff
@@ -99,7 +99,7 @@ def run_train_and_test(
 
 
 def run_train(
-    forecaster: Estimator,
+    forecaster: GluonEstimator,
```

Comment:
There are different `Estimator`s, not just `GluonEstimator`s.

The mypy error is actually helpful. We only want to pass the num_workers argument when it's a `GluonEstimator`.